scripts/PHHY-JP/c101111031.lua

```diff
@@ -0,0 +1,82 @@
+--White Zoa of Dogmatika
+local s,id,o=GetID()
+function s.initial_effect(c)
+	c:EnableReviveLimit()
+	--spsummon condition
+	local e1=Effect.CreateEffect(c)
+	e1:SetType(EFFECT_TYPE_SINGLE)
+	e1:SetProperty(EFFECT_FLAG_CANNOT_DISABLE+EFFECT_FLAG_UNCOPYABLE)
+	e1:SetCode(EFFECT_SPSUMMON_CONDITION)
+	e1:SetValue(s.splimit)
+	c:RegisterEffect(e1)
+	--immune
+	local e2=Effect.CreateEffect(c)
+	e2:SetType(EFFECT_TYPE_FIELD)
+	e2:SetCode(EFFECT_IMMUNE_EFFECT)
+	e2:SetRange(LOCATION_MZONE)
+	e2:SetTargetRange(LOCATION_MZONE,0)
+	e2:SetTarget(aux.TargetBoolFunction(Card.IsSetCard,0x145))
+	e2:SetValue(s.efilter)
+	c:RegisterEffect(e2)
+	--opponent applies effect
+	local e3=Effect.CreateEffect(c)
+	e3:SetDescription(aux.Stringid(id,0))
+	e3:SetCategory(CATEGORY_TOGRAVE+CATEGORY_TOEXTRA)
+	e3:SetType(EFFECT_TYPE_IGNITION)
+	e3:SetRange(LOCATION_MZONE)
+	e3:SetCountLimit(1,id)
+	e3:SetTarget(s.target)
+	e3:SetOperation(s.operation)
+	c:RegisterEffect(e3)
+end
+function s.splimit(e,se,sp,st)
+	return st&SUMMON_TYPE_RITUAL==0 or (se and se:GetHandler():IsSetCard(0x145))
```

Comment:
`You can Ritual Summon this card with "Dogmatikalamity". Cannot be `**`Ritual`**` Summoned, except with a "Dogmatika" card's effect. "Dogmatika" monsters you control are unaffected by your opponent's activated Fusion, Synchro, Xyz, and Link Monsters' effects. During your Main Phase: You can make your opponent choose 1 of these effects.`
`● Send 1 card from their hand and/or Extra Deck to the GY for every 2 cards in their Extra Deck (their choice).`
`● Return all Fusion, Synchro, Xyz, and Link Monsters they control to the Extra Deck.`
`You can only use this effect of "White Zoa of Dogmatika" once per turn.`
Currently, this _must_ be Ritual Summoned, and cannot be Special Summoned by other ways.
```suggestion
	return st&SUMMON_TYPE_RITUAL~=SUMMON_TYPE_RITUAL or (se and se:GetHandler():IsSetCard(0x145))
```